githubのリリースRSS(atom)を読もうとするとエラーになる

<img width="1233" alt="image" src="https://github.com/user-attachments/assets/3ca456af-e320-4a12-bdfa-a544f7b39fea" />

該当ログ

<details>

<summary>WARN</summary>

```
LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
```

</details>

<details>

<summary>ERROR</summary>

```
[ERROR] KeyError: 'published'
Traceback (most recent call last):
  File "/var/task/index.py", line 111, in handler
    add_blog(rss_name, rss_result["entries"], notifier_name, RECENT_PUBLICATION_DAYS)
  File "/var/task/index.py", line 85, in add_blog
    if recently_published(entry["published"], days):
  File "/var/task/feedparser/util.py", line 113, in __getitem__
    return dict.__getitem__(self, key)
```

</details>

--- a/lambda/rss-crawler/index.py
+++ b/lambda/rss-crawler/index.py
@@ -18,14 +18,14 @@ dynamo = boto3.resource("dynamodb")
 table = dynamo.Table(DDB_TABLE_NAME)
 
 
-def recently_published(pubdate, days):
+def recently_published(entry, days):
     """Check if the publication date is recent
 
     Args:
-        pubdate (str): The publication date and time
+        entry (dict): The blog entry
         days (int): The number of days to consider as recent
     """
-
+    pubdate = entry.get("published", entry.get("updated"))
     elapsed_time = datetime.datetime.now() - str2datetime(pubdate)
     print(elapsed_time)
     if elapsed_time.days > days:
@@ -82,12 +82,13 @@ def add_blog(rss_name, entries, notifier_name, days):
     """
 
     for entry in entries:
-        if recently_published(entry["published"], days):
+        pubdate = entry.get("published", entry.get("updated"))
+        if recently_published(entry, days):
             write_to_table(
                 entry["link"],
                 entry["title"],
                 rss_name,
-                str2datetime(entry["published"]).isoformat(),
+                str2datetime(pubdate).isoformat(),
                 notifier_name,
             )
         else:
@@ -103,7 +104,7 @@ def handler(event, context):
         rss_result = feedparser.parse(rss_url)
         print(json.dumps(rss_result))
         print("RSS updated " + rss_result["feed"]["updated"])
-        if not recently_published(rss_result["feed"]["updated"], RECENT_PUBLICATION_DAYS):
+        if not recently_published(rss_result["feed"], RECENT_PUBLICATION_DAYS):
             # Do not process RSS feeds that have not been updated for a certain period of time.
             # If you want to retrieve from the past, change this number of days and re-import.
             print("Skip RSS " + rss_name)

--- a/lambda/rss-crawler/index.py
+++ b/lambda/rss-crawler/index.py
@@ -11,7 +11,7 @@ import dateutil.parser
 # CRAWL_BLOG_URL = json.loads(os.environ["RSS_URL"])
 # NOTIFIERS = json.loads(os.environ["NOTIFIERS"])
 
-RECENT_PUBLICATION_DAYS = 1
+RECENT_PUBLICATION_DAYS = 3
 
 DDB_TABLE_NAME = os.environ["DDB_TABLE_NAME"]
 dynamo = boto3.resource("dynamodb")
@@ -82,12 +82,13 @@ def add_blog(rss_name, entries, notifier_name, days):
     """
 
     for entry in entries:
-        if recently_published(entry["published"], days):
+        published = entry.get("published", entry.get("updated", ""))
+        if recently_published(published, days):
             write_to_table(
                 entry["link"],
                 entry["title"],
                 rss_name,
-                str2datetime(entry["published"]).isoformat(),
+                str2datetime(published).isoformat(),
                 notifier_name,
             )
         else:

--- a/lambda/rss-crawler/index.py
+++ b/lambda/rss-crawler/index.py
@@ -11,7 +11,7 @@ import dateutil.parser
 # CRAWL_BLOG_URL = json.loads(os.environ["RSS_URL"])
 # NOTIFIERS = json.loads(os.environ["NOTIFIERS"])
 
-RECENT_PUBLICATION_DAYS = 3
+RECENT_PUBLICATION_DAYS = 1
 
 DDB_TABLE_NAME = os.environ["DDB_TABLE_NAME"]
 dynamo = boto3.resource("dynamodb")

--- a/lambda/rss-crawler/index.py
+++ b/lambda/rss-crawler/index.py
@@ -18,14 +18,14 @@ dynamo = boto3.resource("dynamodb")
 table = dynamo.Table(DDB_TABLE_NAME)
 
 
-def recently_published(entry, days):
+def recently_published(pubdate, days):
     """Check if the publication date is recent
 
     Args:
-        entry (dict): The blog entry
+        pubdate (str): The publication date and time
         days (int): The number of days to consider as recent
     """
-    pubdate = entry.get("published", entry.get("updated"))
+
     elapsed_time = datetime.datetime.now() - str2datetime(pubdate)
     print(elapsed_time)
     if elapsed_time.days > days:
@@ -82,13 +82,12 @@ def add_blog(rss_name, entries, notifier_name, days):
     """
 
     for entry in entries:
-        pubdate = entry.get("published", entry.get("updated"))
-        if recently_published(entry, days):
+        if recently_published(entry["published"], days):
             write_to_table(
                 entry["link"],
                 entry["title"],
                 rss_name,
-                str2datetime(pubdate).isoformat(),
+                str2datetime(entry["published"]).isoformat(),
                 notifier_name,
             )
         else:
@@ -104,7 +103,7 @@ def handler(event, context):
         rss_result = feedparser.parse(rss_url)
         print(json.dumps(rss_result))
         print("RSS updated " + rss_result["feed"]["updated"])
-        if not recently_published(rss_result["feed"], RECENT_PUBLICATION_DAYS):
+        if not recently_published(rss_result["feed"]["updated"], RECENT_PUBLICATION_DAYS):
             # Do not process RSS feeds that have not been updated for a certain period of time.
             # If you want to retrieve from the past, change this number of days and re-import.
             print("Skip RSS " + rss_name)

